### ISSUE TITLE
select_from_array is marking on option as selected that shouldn't be

### ISSUE THREAD

**Issue by jondcampbell** [Laravel-Backpack/CRUD#58]
When editing an entry my `old($field['name'])` is null and my $key happens to be 0 so the <option> gets marked as selected even though it shouldn't be.

Take a look at https://github.com/Laravel-Backpack/crud/blob/master/src/resources/views/fields/select_from_array.blade.php#L21

I could change my keys for the array so I am not using 0, but would rather not have to.

**Comment by tabacitu** [comment]
Hmm... I believe the problem isn't at that line, but in the HTML element itself: the browser will select the first ```<option>``` in a ```<select>```, to make it easier for you. Your item that has key==0 is the first one, so it gets selected.

**Did you try using the "allows_null" option?** So something like this:

```php
[ // select_from_array
    'name' => 'template',
    'label' => "Template",
    'type' => 'select_from_array',
    'options' => [0 => ‘zero’, 1 => 'One', 2 => ‘Two’],
    'allows_null' => true
],
```
That should put a blank element at the start, so that one gets selected by default.

**Comment by jondcampbell** [comment]
The issue is that that when there is no value in the `old($field['name'])` it returns null and my $key is 0 and Line 21 then essentially evaluates `if (null == 0)` which will return true in php.

The one I want selected gets selected anyway when the loop hits it.
I don't really want to allow nulls as an option.

These are my options for the field:
`'options' => [
    0 => 'Draft',
    1 => 'Published'
],`

I can fix it by changing line 21 to this:
`@if ((isset($field['value']) && $key==$field['value']) || ( ! is_null( old($field['name']) ) && old($field['name']) == $key) )`

But I just am not sure if theres a better way of doing it. We can't use isset because we are checking the value of the old function and not just a plain variable.

**Comment by tabacitu** [comment]
It's now fixed in 2.1 (dev branch, will launch soon).

Thank you!
